python/lsst/daf/butler/logging.py

```diff
@@ -292,18 +292,18 @@ class ButlerLogRecords(_ButlerLogRecords):
     _log_format: str | None = PrivateAttr(None)
 
     @classmethod
-    def from_records(cls, records: Iterable[ButlerLogRecord]) -> "ButlerLogRecords":
+    def from_records(cls, records: Iterable[ButlerLogRecord]) -> Self:
         """Create collection from iterable.
 
         Parameters
         ----------
         records : iterable of `ButlerLogRecord`
             The records to seed this class with.
         """
-        return cls.model_construct(root=list(records))
+        return cast(Self, cls.model_construct(root=list(records)))
```

Comment:
`cls.model_construct()` returns `ButlerLogRecords` and mypy can't work out that's not `Self`.